tests/jballerina-unit-test/src/test/resources/test-src/types/xml/xml-attribute-access-syntax.bal

```diff
@@ -1,11 +1,44 @@
 import ballerina/lang.'xml;
 
-function getElementAttrBasic() returns string|error? {
+type XMLElement xml:Element;
+type XMLElement2 xml<xml:Element>;
+type XMLElement3 xml<xml<xml:Element>>;
+
+function getElementAttrBasic() returns error? {
     xml x = xml `<root attr="attr-val"><a></a><b></b></root>`;
-    string|error? val = x.attr;
-    return val;
+    xml<xml:Element> x2 = xml `<root attr="attr-val"><a></a><b></b></root>`;
+    xml<xml<xml:Element>> x3 = xml `<root attr="attr-val"><a></a><b></b></root>`;
+    assert(check x.attr, "attr-val");
+    assert(check x2.attr, "attr-val");
+    assert(check x3.attr, "attr-val");
+}
+
+function getOptionalElementAttrBasic() returns error? {
+    xml:Element x = xml `<elem xmlns="ns-uri" attr="attr-val" xml:space="default"></elem>`;
+    xml<xml:Element> x2 = xml `<elem xmlns="ns-uri" attr="attr-val" xml:space="default"></elem>`;
+    xml<xml<xml:Element>> x3 = xml `<elem xmlns="ns-uri" attr="attr-val" xml:space="default"></elem>`;
+    assert(check x?.attr, "attr-val");
```

Comment:
```suggestion
    assert(x?.attr, "attr-val");
```

We don't need to use `check` since it is optional field access
Check other places too